packages/host/src/lib/serviceDiscovery/topic.ts

```diff
@@ -41,18 +52,40 @@ export class Topic extends Duplex implements TopicHandler {
     }
     origin() { return this._origin; }
 
-    _read(_size: number): void { }
-    _write(chunk: any, encoding: BufferEncoding, callback: (error?: Error | null | undefined) => void): void {
+    acceptPipe(rdble: Readable) {
+        this._pipeQueue.push(rdble);
+        this.consumePipe();
+    }
+
+    consumePipe() {
```

Comment:
This looks like it shouldn't be here. Duplicates logic from post topic where we call `.on("end")` anyway.

Reply:
Not really - this can consume multiple piped streams one after another - it's a cleaner way of consuming the whole stream, and the `on("end")` is actually on the pipe, not on the `topic`.